src/EditorFeatures/CSharpTest/AddMissingImports/CSharpAddMissingImportsFeatureServiceTests.cs

```diff
@@ -0,0 +1,151 @@
+        {
+            var code = @"
+class C
+{
+    public [|D|] Foo { get; }
+}
+
+namespace A
+{
+    public class D { }
+}
+
+namespace B
+{
+    public class D { }
+}
+";
+
+            await AssertDocumentUnchangedAsync(code).ConfigureAwait(false);
+        }
+
+        private async Task AssertDocumentUnchangedAsync(string initialMarkup)
+        {
+            using (var workspace = TestWorkspace.CreateCSharp(initialMarkup))
+            {
+                var diagnosticAnalyzerService = InitializeDiagnosticAnalyzerService(workspace);
+
+                var addMissingImportsService = new CSharpAddMissingImportsFeatureService(diagnosticAnalyzerService);
+
+                var hostDocument = workspace.Documents.First();
+                var documentId = hostDocument.Id;
+                var textSpan = hostDocument.SelectedSpans.First();
+
+                var document = workspace.CurrentSolution.GetDocument(documentId);
+
+                var newProject = await addMissingImportsService.AddMissingImportsAsync(document, textSpan, CancellationToken.None).ConfigureAwait(false);
+                var newDocument = newProject.GetDocument(documentId);
+
+                Assert.Equal(document, newDocument);
+            }
+        }
+
+        private async Task AssertDocumentChangedAsync(string initialMarkup, string expectedMarkup)
+        {
+            using (var workspace = TestWorkspace.CreateCSharp(initialMarkup))
+            {
+                var diagnosticAnalyzerService = InitializeDiagnosticAnalyzerService(workspace);
+
+                var addMissingImportsService = new CSharpAddMissingImportsFeatureService(diagnosticAnalyzerService);
+
+                var hostDocument = workspace.Documents.First();
+                var documentId = hostDocument.Id;
+                var textSpan = hostDocument.SelectedSpans.First();
+
+                var document = workspace.CurrentSolution.GetDocument(documentId);
+
+                var newProject = await addMissingImportsService.AddMissingImportsAsync(document, textSpan, CancellationToken.None).ConfigureAwait(false);
+                var newDocument = newProject.GetDocument(documentId);
+
+                Assert.NotEqual(document, newDocument);
+
+                var text = await newDocument.GetTextAsync().ConfigureAwait(false);
+
+                Assert.Equal(expectedMarkup, text.ToString());
+            }
+        }
+
+        private IDiagnosticAnalyzerService InitializeDiagnosticAnalyzerService(Workspace workspace)
+        {
+            var diagnosticAnalyzer = DiagnosticExtensions.GetCompilerDiagnosticAnalyzer(LanguageName);
+            var exceptionDiagnosticsSource = new TestHostDiagnosticUpdateSource(workspace);
+
+            var diagnosticAnalyzerService = new TestDiagnosticAnalyzerService(LanguageName, diagnosticAnalyzer, exceptionDiagnosticsSource);
+            diagnosticAnalyzerService.CreateIncrementalAnalyzer(workspace);
+
+            return diagnosticAnalyzerService;
```

Comment:
I dont see any test that actually add multiple missing usings? should we have at least some cover of fix all/many import cases?